cranelift/codegen/src/isa/x64/inst/emit.rs

```diff
@@ -1998,18 +1998,34 @@ pub(crate) fn emit(
             src2,
             dst,
         } => {
+            use LegacyPrefixes as LP;
+            use OpcodeMap as OM;
+
             let dst = allocs.next(dst.to_reg().to_reg());
-            let src1 = allocs.next(src1.to_reg());
-            let src2 = src2.clone().to_reg_mem_imm().with_allocs(allocs);
+            let (src1, src2) = if inst.produces_const() {
```

Comment:
I'm not wild about the argument-dependent regalloc metadata shape here, and ideally I'd prefer us to split out variants like `XmmRmiRVexConst` that are used instead whenever the intent of an instruction is to produce a constant 0/1/whatever, and that assert the correct setup for that; but that's a separate refactor you don't necessarily need to do in this PR. Just noting as I see the existing pattern expanding...

Reply:
Sure yeah, I was just following the preexisting pattern for the xmm instructions, but I can try to update those too in a subsequent PR.